### ISSUE TITLE
SSL Certification Renewal failure

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-online-tools#266]
The Certbot SSL certificate renewal is failing with the following message:

```
1 renew failure(s), 0 parse failure(s)
IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: tools.spdx.org
   Type:   unauthorized
   Detail: Invalid response from
   https://tools.spdx.org/.well-known/acme-challenge/1R0oTS8-1WBIYOX4yi2hFbB3ZBPkBMfeCIBYriNviZc
   [52.32.53.255]: "<!DOCTYPE html>\n\n\n<html lang=\"en\">\n
   <head>\n    <meta charset=\"utf-8\">\n    <meta
   http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">"

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address.

```

The domain name looks correct and the IP address looks correct.

**Comment by goneall** [comment]
@swinslow @Ugtan @rtgdk Let me know if you have any ideas on this one.

The script just does a certbot renew: https://github.com/spdx/spdx-online-tools/blob/5e6a14c38c416a2a48c1a752f03afe8f57352070/docker-compose.prod.yml#L25

This is somewhat urgent as the certificate is set to expire by Dec. 10th.

One thing I could do is re-issue a new certificate using the script: https://github.com/spdx/spdx-online-tools/blob/master/scripts/init-letsencrypt.sh

**Comment by swinslow** [comment]
I've never set up certbot using docker or docker-compose -- when I've used certbot in the past it has been by following the [DigitalOcean instructions](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04) which assume you're on a virtual machine.

Looking at the error though, it looks like:
* the Let's Encrypt validation server is trying to query a "well-known" subpage that certbot should have access to create (this is the `https://tools.spdx.org/.well-known/acme-challenge/...` URL in the response)
* but when it queries that URL, it isn't getting a page created by certbot -- instead it's getting the beginning of the SPDX Online Tools HTML content
* which suggests that certbot isn't configured correctly to be able to create that .well-known/... URL

[This](https://certbot.eff.org/docs/using.html#webroot) looks like it _might_ be the certbot method that's being used here.

Looking at a random instructions site [here](https://medium.com/@pentacent/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71) -- do a Ctrl-F search for "Now we can make nginx serve". It looks like there is a section that should go in the data/nginx/app.conf file to create a location for the .well-known URL:

```
location /.well-known/acme-challenge/ {
    root /var/www/certbot;
}
```

Is that maybe what's needed here? I don't see that section in the existing app.conf file so maybe it got skipped accidentally?

**Comment by goneall** [comment]
Resolved by PR #269
